core/src/main/java/hudson/model/Queue.java

```diff
@@ -1219,6 +1219,59 @@ public static void withLock(Runnable runnable) {
         }
     }
 
+    /**
+     * Some operations require to be performed with the {@link Queue} lock held. Use one of these methods rather
+     * than locking directly on Queue in order to allow for future refactoring.
+     * @param runnable the operation to perform.
+     * @return {@code true} if the lock available and the operation performed. 
+     * @since 1.FIXME
+     */
+    public static boolean tryWithLock(Runnable runnable) {
+        final Jenkins jenkins = Jenkins.getInstance();
+        final Queue queue = jenkins == null ? null : jenkins.getQueue();
+        if (queue == null) {
```

Comment:
NIT: it should not really happen (?). Otherwise, we still have a risk of surprising behaviors on Jenkins startup